core/trino-spi/src/main/java/io/trino/spi/connector/ColumnMetadata.java

```diff
@@ -94,7 +93,8 @@ public ColumnMetadata(String name, Type type, boolean nullable, String comment,
         this.comment = comment;
         this.extraInfo = extraInfo;
         this.hidden = hidden;
-        this.properties = properties.isEmpty() ? emptyMap() : unmodifiableMap(new LinkedHashMap<>(properties));
+        // Avoiding doing a copy as properties can be lazy
+        this.properties = properties.isEmpty() ? emptyMap() : unmodifiableMap(properties);
```

Comment:
after this change, `ColumnMetadata` is no longer immutable

Reply:
That is the price.

I don't have anything better instead. Anything like `Supplier<Map<>>` is not much better other than that it is clear.

Reply:
Not only that, but the class goes from becoming a simple container to being a “service” object. What are the guarantees about the lazy behavior? Can it do I/O or use network resources, can it block for an extended period, etc.

I don’t think we want to go in this direction.